.github/workflows/deploy_and_test.yml

```diff
@@ -0,0 +1,26 @@
+name: CI
+
+on:
+  workflow_dispatch:
+  push:
+      branches: [ main ]
+  pull_request:
+      branches: [ main ]
+
+jobs:
+  build-and-test:
```

Comment:
Checkout https://github.com/ITISFoundation/osparc-simcore-python-client/blob/f5f297b9bf3c15010ae1a06ced9c8e979f092678/.github/workflows/test.yml

notice:
 - using ``actions/setup-python@v2`` 
 - targets multiple python version ``strategy.matrix``
 - uses cache
It is actually a basic extension of  [github actions tutotirla for python](https://docs.github.com/actions/automating-builds-and-tests/building-and-testing-nodejs-or-python?langId=py)

Reply:
Using the setup-python action